be/src/env/env.h

```diff
@@ -217,6 +217,9 @@ class RandomAccessFile {
     RandomAccessFile() = default;
     virtual ~RandomAccessFile() = default;
 
+    virtual Status open() { return Status::OK(); }
+    virtual void close() {}
```

Comment:
should called automatically in destructor

Reply:
Sometimes file handlers are hold by the scan node, and deleted at the destructor of scan node.  But during the scan ,if we are at EOF, we can close and release the resources hold by this file handle ASAP.

Reply:
It's ok to do early close, but we have to check if it is closed in destructor

Reply:
OK. Will fix that.

Reply:
So we have `close` but no `open`?

Reply:
I think we can remove `close`, if we wanna perform close early, we can `down_cast` it

Reply:
But was that  be awkward? We have `open` but no `close`?